### PR TITLE
Dependency updates for master

### PR DESCRIPTION
**Pull request by Lex-2008**
Update openldap from 2.4.50 to 2.4.53
Update libcurl-hub from 7.70.0 to 7.72.0
Update apache from 2.4.43 to 2.4.46
Update git from 2.27.0 to 2.28.0
Update postgresql from 12.3 to 12.4
Update php from 7.4.6 to 7.4.10
Update libcurl from 7.70.0 to 7.72.0

### CODE REVIEW

**Comment by Lex-2008** [comment]
@cf-bottom exotics build please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/5615/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/5615/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-5615/

**Comment by Lex-2008** [comment]
green-ish build: https://ci.cfengine.com/job/pr-pipeline/5641/

File: deps-packaging/openldap/cfbuild-openldap.spec [hunk 1]
```diff
@@ -96,3 +96,4 @@ CFEngine Build Automation -- openldap -- development files
 
 
 
+
```
Comment (vpodzime) [inline deps-packaging/openldap/cfbuild-openldap.spec:99]: Why do we keep adding blank lines here?

File: deps-packaging/rsync/cfbuild-rsync.spec [hunk 2]
```diff
@@ -18,7 +18,10 @@ AutoReqProv: no
 mkdir -p %{_builddir}
 %setup -q -n rsync-%{rsync_version}
 
-./configure --prefix=%{prefix} --with-included-zlib=%{prefix}
+# liblz4, libxxhash, libzstd, and libssl give rsync extra compression
+# algorithms, extra checksum algorithms, and allow use of openssl's crypto lib
+# for (potentially) faster MD4/MD5 checksums.
+./configure --prefix=%{prefix} --with-included-zlib=%{prefix} --disable-openssl --disable-xxhash --disable-zstd --disable-lz4
```
Comment (vpodzime) [inline deps-packaging/rsync/cfbuild-rsync.spec:24]: Why `--disable-openssl` when we build and ship `openssl`?

Comment (vpodzime) [reply deps-packaging/rsync/cfbuild-rsync.spec:24]: The potentially faster checksums could happen in FR.

**Review by vpodzime — COMMENTED** [review]
Looks good to me otherwise.

File: deps-packaging/openldap/cfbuild-openldap.spec (hunk 1 above)
Comment (Lex-2008) [reply deps-packaging/openldap/cfbuild-openldap.spec:99]: good point!
https://github.com/cf-bottom/self/pull/57

File: deps-packaging/rsync/cfbuild-rsync.spec (hunk 2 above)
Comment (Lex-2008) [reply deps-packaging/rsync/cfbuild-rsync.spec:24]: good point! What about RHEL8? Do you think it's ok to link to system openssl libs?

Comment (vpodzime) [reply deps-packaging/rsync/cfbuild-rsync.spec]: Sure, should be fine.

File: deps-packaging/rsync/cfbuild-rsync.spec [hunk 3]
```diff
@@ -21,7 +21,8 @@ mkdir -p %{_builddir}
 # liblz4, libxxhash, libzstd, and libssl give rsync extra compression
 # algorithms, extra checksum algorithms, and allow use of openssl's crypto lib
 # for (potentially) faster MD4/MD5 checksums.
-./configure --prefix=%{prefix} --with-included-zlib=%{prefix} --disable-openssl --disable-xxhash --disable-zstd --disable-lz4
+find %{prefix} -name 'md5.h'
+./configure --prefix=%{prefix} --with-included-zlib=%{prefix} CFLAGS="-I%{prefix}/include" --disable-xxhash --disable-zstd --disable-lz4
```
Comment (vpodzime) [inline deps-packaging/rsync/cfbuild-rsync.spec:25]: Hmm, I would have expected `-I%{prefix}/include` to be working by default. Anyways, please change this to `CPPFLAGS`, `-I` is an option for the pre-processor not the compiler.

File: deps-packaging/rsync/cfbuild-rsync.spec (hunk 2 above)
Comment (Lex-2008) [reply deps-packaging/rsync/cfbuild-rsync.spec]: on the other side, we build rsync only on HUB, and we don't build HUB for RHEL8 - so we don't need to fix this right now :P

**Comment by Lex-2008** [comment]
rather green build: https://ci.cfengine.com/job/pr-pipeline/5685/
@vpodzime : ok to merge?

File: deps-packaging/rsync/cfbuild-rsync.spec [hunk 4]
```diff
@@ -21,7 +21,8 @@ mkdir -p %{_builddir}
 # liblz4, libxxhash, libzstd, and libssl give rsync extra compression
 # algorithms, extra checksum algorithms, and allow use of openssl's crypto lib
 # for (potentially) faster MD4/MD5 checksums.
-./configure --prefix=%{prefix} --with-included-zlib=%{prefix} --disable-openssl --disable-xxhash --disable-zstd --disable-lz4
+find %{prefix} -name 'md5.h'
```
Comment (vpodzime) [inline deps-packaging/rsync/cfbuild-rsync.spec:24]: Drop this `find`.

**Review by vpodzime — COMMENTED** [review]
Please add some explanation to the last commit.

File: deps-packaging/rsync/cfbuild-rsync.spec (hunk 4 above)
Comment (Lex-2008) [reply deps-packaging/rsync/cfbuild-rsync.spec:24]: good :eyes: !

**Comment by Lex-2008** [comment]
right,  last commit should be merged into rsync one

File: deps-packaging/rsync/cfbuild-rsync.spec (hunk 2 above)
Comment (vpodzime) [reply deps-packaging/rsync/cfbuild-rsync.spec]: Too late now :tongue:

File: deps-packaging/rsync/cfbuild-rsync.spec [hunk 5]
```diff
@@ -31,6 +34,7 @@ rm -rf ${RPM_BUILD_ROOT}
 make install DESTDIR=${RPM_BUILD_ROOT}
 
 rm -rf ${RPM_BUILD_ROOT}%{prefix}/share/
+rm ${RPM_BUILD_ROOT}%{prefix}/bin/rsync-ssl
```
Comment (vpodzime) [inline deps-packaging/rsync/cfbuild-rsync.spec:37]: Do we want to remove or package this now that we are building with OpenSSL support?

File: deps-packaging/rsync/cfbuild-rsync.spec [hunk 6]
```diff
@@ -18,7 +18,10 @@ AutoReqProv: no
 mkdir -p %{_builddir}
 %setup -q -n rsync-%{rsync_version}
 
-./configure --prefix=%{prefix} --with-included-zlib=%{prefix}
+# liblz4, libxxhash, libzstd, and libssl give rsync extra compression
+# algorithms, extra checksum algorithms, and allow use of openssl's crypto lib
+# for (potentially) faster MD4/MD5 checksums.
+./configure --prefix=%{prefix} --with-included-zlib=%{prefix} CPPFLAGS="-I%{prefix}/include" --disable-xxhash --disable-zstd --disable-lz4
```
Comment (vpodzime) [inline deps-packaging/rsync/cfbuild-rsync.spec:24]: Let's see if this works with on RHEL 8 hub now.

File: deps-packaging/rsync/source [hunk 7]
```diff
@@ -1 +1 @@
-http://mirrors.ibiblio.org/rsync/src/
+https://download.samba.org/pub/rsync/src/
```
Comment (vpodzime) [inline deps-packaging/rsync/source:1]: Removed newline, please keep it there.

File: deps-packaging/rsync/cfbuild-rsync.spec (hunk 5 above)
Comment (Lex-2008) [reply deps-packaging/rsync/cfbuild-rsync.spec:37]: `rsync-ssl` is just a script which wraps `rsync` into `stunnel`, according to its man page:
https://download.samba.org/pub/rsync/rsync-ssl.1
Its usage doesn't depend on us linking `rsync` against openssl libs, and it doesn't look like we need it :)

**Comment by Lex-2008** [comment]
rebased on top of recent master. @cf-bottom : build it please, with exotics

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/5695/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/5695/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-5695/

**Comment by vpodzime** [comment]
From the *PACKAGES_HUB_x86_64_linux_redhat_8* build:

```
15:34:05 checking whether to enable use of openssl crypto library... yes
15:34:05 checking for library containing MD5_Init... -lcrypto
```
:tada:

**Review by vpodzime — APPROVED** [review]
Good job, @Lex-2008! ACK

**Pull request merged**